pkg/apis/projectcalico/v3/packetcapture.go

```diff
@@ -69,13 +69,12 @@ type PacketCaptureRule struct {
 	// a specific IP protocol.
 	//
 	// Must be one of these string values: "TCP", "UDP", "ICMP", "ICMPv6", "SCTP", "UDPLite"
-	// or an integer in the range 1-255.
 	Protocol *numorstring.Protocol `json:"protocol,omitempty" validate:"omitempty"`
 
 	// Ports is an optional field that defines a filter for all traffic that has a
 	// source or destination port that matches one of these ranges/values. This value is a
 	// list of integers or strings that represent ranges of ports.
-	Ports []numorstring.Port `json:"ports,omitempty" validate:"omitempty,dive"`
+	Ports []numorstring.NumPort `json:"ports,omitempty" validate:"omitempty,dive"`
```

Comment:
One simpler option might be to just validate that the user has not set named ports here.  Then it feels a little more future-proofed (and doesn't require new types).

Reply:
I was thinking about that. If we ever want to support this, then yes, I agree. If no, I think the API definition might be a better place than a runtime error.

Reply:
I am doing the same for protocols, as in I created a validator to reject numerical values. I will simply update the descriptions on the fields to reflect this then.